docs/proposals/21-global-failover.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Global Failover
+gep-number: 21
+creation-date: 2022-07-12
+status: implementable
+authors:
+- "@jia-jerry"
+- "@EmoinLanyu"
+reviewers:
+- "@main-reviewer-1"
+- "@main-reviewer-2"
+---
+
+# GEP-21: Fail Over Multi-AZ Services by Isolating Unhealthy AZ
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+    - [New Changes](#new-changes)
+    - [Control Failover / Failback Process](#control-failover--failback-process)
+    - [Failover Workflow](#failover-workflow)
+    - [Failback Workflow](#failback-workflow)
+- [Future Plan](#future-plan)
+
+## Summary
+
+This `GEP` introduces an operating procedure allowing for Gardener operators to isolate an unhealthy AZ of a `Shoot` cluster, and consequently fail over services running inside to healthy AZs. This procedure consists of two steps: removing `MachineDeployment` in the unhealthy AZ, and applying `SubnetIsolation` to the unhealthy AZ. The `SubnetIsolation` is a new extension resource that is reconciled by the respective provider extension controller.
+
+Moreover, it covers a failback procedure for services with multi-AZ setup once the unhealthy AZ is recovered.
+
+## Motivation
+
+Several outages in past years have shown that AZ-specific degradations / failures in hyperscaler services result in significant issues for services. There are multiple patterns of known issues.
+
+1. Some services operate multiple instances of each components redundantly within a single AZ. This will likely provide uptimes approaching 99.5%, but such services are subject to AZ failures and remain offline until the underlying AZ is restored.
```

Comment:
If components run already in multiple zones, why are they written in a way that the "services remain offline", if a zone goes down? The whole purpose of such an implementation is exactly the opposite, i.e. to stay up. That's how we do it and how hopefully everybody deals with it.

Reply:
Absolute. Global failover won't handle pattern 1 issues. Instead, the service team should make services multi-AZ ready. We make such conclusion in the following paragraph.
> However, we shouldn't fail over a Shoot cluster with single AZ HA services.

It might be better to move this sentence into pattern 1 section.